### PR TITLE
DAOS-13627 build: Enable and fix issues with -Wextra.

### PR DESCRIPTION
**Pull request by ashleypittman**
Make use of the static analysis that modern compiers can
provide and fix issues in the codebase.
Leave a couple of warnings disabled that result in multiple errors

Required-githooks: true

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/container/srv_container.c
Comment (ashleypittman) [inline src/container/srv_container.c:4220]: This one I'm not sure about but pcf_parts appears to be a pointer, in which case I think this is the correct change.

File: src/include/daos_obj_class.h
Comment (ashleypittman) [inline src/include/daos_obj_class.h:1]: This file was interesting, we have a couple of functions that take `enum daos_obj_redun` as an option and they only accepted the core types as the complex ones were a different, anonymous enum.  This change makes all values the same type of enum.

File: .github/workflows/landing-builds.yml
Comment (ashleypittman) [inline .github/workflows/landing-builds.yml:21]: Whitespace change to trigger landings testing for this PR to get full compiler coverage.  I can remove before landing.

File: src/container/srv_container.c
Comment (daltonbohning) [reply src/container/srv_container.c:4220]: If it's a pointer, then should it check NULL instead of 0?

**Review by daltonbohning — DISMISSED** [review]

File: src/container/srv_container.c
Comment (ashleypittman) [reply src/container/srv_container.c:4220]: Will change (and the same change in dfs) when I re-push to remove the change to landings.yaml.

**Comment by ashleypittman** [comment]
This just fixes the issues for now but doesn't enable the flag, that is blocked by the changes @daltonbohning is making to fix some coverity issues.

**Review by jolivier23 — DISMISSED** [review]
Why is -Wextra not added anywhere?  Or is this just making it possible to use it?

**Comment by jolivier23** [comment]
> Why is -Wextra not added anywhere? Or is this just making it possible to use it?

nevermind, just saw the comment above.

File: .github/workflows/landing-builds.yml
Comment (jolivier23) [reply .github/workflows/landing-builds.yml]: Does it not trigger testing if you just do --allow-empty commit ?

Comment (ashleypittman) [reply .github/workflows/landing-builds.yml]: GitHub actions only tests the full matrix of compilers for landings builds, not PRs.

**Comment by ashleypittman** [comment]
https://github.com/daos-stack/daos/pull/12319 is the other PR that's needed to enable the flag.

**Comment by daltonbohning** [comment]
> #12319 is the other PR that's needed to enable the flag.

Just merged

**Review by jolivier23 — DISMISSED** [review]

**Review by daltonbohning — DISMISSED** [review]

**Review by daltonbohning — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**
